homeassistant/components/nissan_leaf.py

```diff
@@ -0,0 +1,467 @@
+                if attempt > 0:
+                    _LOGGER.debug("Climate data not in yet. (%s) (%s). "
+                                  "Waiting 5 seconds", self.leaf.vin, attempt)
+                    await asyncio.sleep(5)
+
+                climate_result = await self.hass.async_add_job(
+                    self.leaf.get_stop_climate_control_result, request
+                )
+
+                if climate_result is not None:
+                    break
+
+        if climate_result is not None:
+            _LOGGER.debug("Climate result: %s", climate_result.__dict__)
+            async_dispatcher_send(self.hass, SIGNAL_UPDATE_LEAF)
+            return climate_result.is_hvac_running == toggle
+
+        _LOGGER.debug("Climate result not returned by Nissan servers")
+        return False
+
+    async def async_get_location(self):
+        """Get location from Nissan servers."""
+        request = await self.hass.async_add_job(self.leaf.request_location)
+        for attempt in range(MAX_RESPONSE_ATTEMPTS):
+            if attempt > 0:
+                _LOGGER.debug("Location data not in yet. (%s) (%s). "
+                              "Waiting 5 seconds", self.leaf.vin, attempt)
+                await asyncio.sleep(5)
+
+            location_status = await self.hass.async_add_job(
+                self.leaf.get_status_from_location, request
+            )
+
+            if location_status is not None:
+                _LOGGER.debug("Location_status=%s", location_status.__dict__)
+                break
+
+        async_dispatcher_send(self.hass, SIGNAL_UPDATE_LEAF)
+        return location_status
+
+    async def async_start_charging(self):
+        """Request start charging via Nissan servers."""
+        # Send the command to request charging is started to Nissan servers.
+        # If that completes OK then trigger a fresh update to pull the
+        # charging status from the car after waiting a minute for the
+        # charging request to reach the car.
+        result = await self.hass.async_add_job(self.leaf.start_charging)
+        if result:
+            _LOGGER.debug("Start charging sent, "
+                          "request updated data in 1 minute")
+            check_charge_at = utcnow() + timedelta(minutes=1)
+            async_track_point_in_utc_time(
+                self.hass, self.async_refresh_data, check_charge_at)
+
+
+class LeafEntity(Entity):
+    """Base class for Nissan Leaf entity."""
+
+    def __init__(self, car):
+        """Store LeafDataStore upon init."""
+        self.car = car
+
+    def log_registration(self):
+        """Log registration."""
+        _LOGGER.debug(
+            "Registered %s component for VIN %s",
+            self.__class__.__name__, self.car.leaf.vin)
+
+    @property
+    def device_state_attributes(self):
+        """Return default attributes for Nissan Leaf entities."""
+        return {
+            'homebridge_serial': self.car.leaf.vin,
```

Comment:
what are these attributes for ?

Reply:
@BenWoodford Are you able to help with this one.  I've not got a homebridge, so I'm not sure.

Reply:
we'll if this is for integration with home kit, this simply does not belong in the nissan leaf component.

Reply:
Removed homebridge_* state attributes in commit https://github.com/home-assistant/home-assistant/pull/19786/commits/cf0b761be652865a4bc96c53000addd29d53b0a2